### PR TITLE
Roll clang to llvmorg-15-init-1464-gbf7f8d6f

### PR DESCRIPTION
**Pull request by jonathanmetzman**
This is incremental roll meant to break less than #8108

### CODE REVIEW

**Comment by jonathanmetzman** [comment]
/gcbrun trial_build.py all --sanitizer address memory undefined coverage --fuzzing-engine libfuzzer afl honggfuzz

**Comment by jonathanmetzman** [comment]
these projects are all broken due to the same issue in cryptofuzz:

bearssl
bitcoin-core
circl
libecc
libressl
nettle
num-bigint
relic
rustcrypto
simd
wolfssl

I will hold back their images so they don't get to clang-15 just yet.
@guidovranken Can you fix this and let me know so we can upgrade these?

**Comment by jonathanmetzman** [comment]
/gcbrun trial_build.py skcms --sanitizer address memory undefined coverage --fuzzing-engine libfuzzer afl honggfuzz

**Comment by guidovranken** [comment]
I don't get this error with

```
$ clang++-15 --version
Ubuntu clang version 15.0.0-++20220815123643+7af201cba581-1~exp1~20220815123735.23
```

Mind giving the exact $CXX $CXXFLAGS ?

**Comment by jonathanmetzman** [comment]
> 

Might be better to investigate this tomorrow when you can test easily by just unpinning base-builder.

**Comment by jonathanmetzman** [comment]
/gcbrun trial_build.py all --sanitizer address memory undefined coverage --fuzzing-engine libfuzzer afl honggfuzz

**Comment by jonathanmetzman** [comment]
/gcbrun trial_build.py all --sanitizer address memory undefined coverage --fuzzing-engine libfuzzer afl honggfuzz

**Comment by jonathanmetzman** [comment]
The builds failures are acceptable to me

**Review by oliverchang — APPROVED** [review]

File: projects/bitcoin-core/Dockerfile [hunk 1]
```diff
@@ -14,7 +14,9 @@
 #
 ################################################################################
 
-FROM gcr.io/oss-fuzz-base/base-builder-rust
+# Held back because of github.com/google/oss-fuzz/pull/8313
+# Please fix failure and upgrade.
```
Comment (MarcoFalke) [inline projects/bitcoin-core/Dockerfile:18]: ACK, we'll investigate after merge

**Review by MarcoFalke — APPROVED** [review]
Thanks for the bump. Hopefully this will allow us to drop the two workarounds for various clang-14 bugs

**Comment by jonathanmetzman** [comment]
Other failures I don't plan on fixing/pinning:
spidermonkey-ufi
lldb-eval
inchi
assimp

**Pull request merged**